receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go

```diff
@@ -114,3 +117,23 @@ func getProcessHandlesInternal() (processHandles, error) {
 
 	return &gopsProcessHandles{handles: processes}, nil
 }
+
+func parentPid(handle processHandle, pid int32) (int32, error) {
+	// special case for pid 0
+	if pid == 0 {
+		return 0, nil
+	}
+	parent, err := handle.Parent()
+
+	if err != nil {
+		// return pid of -1 along with error for all other problems retrieving parent pid
+		return -1, err
+	}
+
+	// if a process does not have a parent return the pid of the process as it's own parent
```

Comment:
this should not happen but gops does report some processes as having no parent.

Reply:
Why returning process's PID as parent PID? Is this some adopted fallback?

Reply:
Yes, just using it as a fallback.  I don't think it's serious enough of an error to report or fail the scrape.

Reply:
Shouldn't it be `0` as it mostly represents no parent AFAIR?

Reply:
Technically speaking every process should have a parent.  If no immediate relative is found then pid 1 (init process) becomes the parent.  I can set it to 0 though.  Would like to avoid using 1 so that we can distinguish between processes where the parent is clearly identified vs processes where we were not able to identify the parent.